reactivesocket-core/src/main/java/io/reactivesocket/exceptions/Exceptions.java

```diff
@@ -30,9 +30,15 @@ public static Throwable from(Frame frame) {
         final int errorCode = Frame.Error.errorCode(frame);
         String message = "<empty message>";
         final ByteBuffer byteBuffer = frame.getMetadata();
-        if (byteBuffer.hasArray()) {
```

Comment:
hasArray here seems problematic in two ways.

1) assuming array backed
2) assuming that the byte buffer is not a sliced with a shared larger backing array

package io.reactivesocket.exceptions;

```
import java.nio.ByteBuffer;

public class AAAA {
    public static void main(String[] args) {
        ByteBuffer b = ByteBuffer.allocate(100);
        b.position(20).limit(50);
        ByteBuffer b2 = b.slice();

        System.out.println(b.capacity()); // 100
        System.out.println(b.remaining()); // 30
        System.out.println(b.array()); // [B@d716361
        System.out.println(b.array().length); // 100

        System.out.println();

        System.out.println(b2.capacity()); // 30
        System.out.println(b2.remaining()); // 30
        System.out.println(b2.array()); // [B@d716361
        System.out.println(b2.array().length); // 100
    }
}
```